core/src/main/java/com/github/jsonldjava/core/JsonLdProcessor.java

```diff
@@ -319,14 +318,18 @@ public static Object flatten(Object input, JsonLdOptions opts) throws JsonLdErro
         // to a new empty
         // context, otherwise.
         final JsonLdApi api = new JsonLdApi(expandedInput, opts);
-        final Context activeCtx = api.context
-                .parse(((Map<String, Object>) frame).get(JsonLdConsts.CONTEXT));
+        final Object context = ((Map<String, Object>) frame).get(JsonLdConsts.CONTEXT);
+        final Context activeCtx = api.context.parse(context);
         final List<Object> framed = api.frame(expandedInput, expandedFrame);
         if (opts.getPruneBlankNodeIdentifiers()) {
             JsonLdUtils.pruneBlankNodes(framed);
         }
         Object compacted = api.compact(activeCtx, null, framed, opts.getCompactArrays());
-        final Map<String, Object> rval = activeCtx.serialize();
+        final Map<String, Object> rval = newMap();
+        final Object returnedContext = returnedContext(context, opts);
+        if(returnedContext != null) {
+            rval.put(JsonLdConsts.CONTEXT, context);
```

Comment:
I am not exactly sure, but instead of putting 'context' it seems that 'returnedContext' would be the proper Object (although the tests don't cover this - both Objects in the tests are always identically).